guest/internal/prefilter/prefilter_exports.go

```diff
@@ -49,6 +49,8 @@ func _prefilter() uint32 { //nolint
 	// This function begins a new scheduling cycle: zero out any cycle state.
 	currentPod = nil
 	currentCycleState = map[string]any{}
+	currentNodeInfoList = nil
+	isFullNodeInfoList = false
```

Comment:
Why does this `isFullNodeInfoList` logic work? Is it because the Node used in the scheduler's step always goes through the prefilter process?

Reply:
Yes, we call preFilter regardless of whether an user implements preFilter or not, so that we can reset all stuff the start of the scheduling cycle.